### PR TITLE
Move Regional PD to GA

### PR DESCRIPTION
**Pull request by jingxu97**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:
Move GCERegionalPersistentDisk feature to GA

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # #59988

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
GCERegionalPersistentDisk feature is GA now!
```

### CODE REVIEW

**Comment by jingxu97** [comment]
cc @saad-ali @verult @davidz627

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ const (
 	CRIContainerLogRotation utilfeature.Feature = "CRIContainerLogRotation"
 
 	// owner: @verult
-	// beta: v1.10
+	// GA: v1.10
```
Comment (davidz627) [inline pkg/features/kube_features.go]: should this version be updated for GA

File: pkg/cloudprovider/providers/gce/gce_disks_test.go [hunk 2]
```diff
@@ -723,9 +723,9 @@ func (manager *FakeServiceManager) CreateRegionalDiskOnCloudProvider(
 	tagsStr string,
 	diskType string,
 	zones sets.String) error {
-	manager.createDiskCalled = true
-	diskTypeURI := gceComputeAPIEndpointBeta + "projects/" + fmt.Sprintf(diskTypeURITemplateRegional, manager.gceProjectID, manager.gceRegion, diskType)
 
+	manager.createDiskCalled = true
+	diskTypeURI := gceComputeAPIEndpoint + "projects/" + fmt.Sprintf(diskTypeURITemplateRegional, manager.gceProjectID, manager.gceRegion, diskType)
 	switch t := manager.targetAPI; t {
```
Comment (verult) [inline pkg/cloudprovider/providers/gce/gce_disks_test.go]: Now that all calls to the GCE PD API are GA, can we remove the checks for other versions?

**Review by verult — COMMENTED** [review]
Note for reviewers: `gen.go` and `gen_test.go` are both automatically generated.

File: pkg/cloudprovider/providers/gce/gce_disks_test.go (hunk 2 above)
Comment (jingxu97) [reply pkg/cloudprovider/providers/gce/gce_disks_test.go]: fixed

**Comment by jingxu97** [comment]
@davidz @verult comments are addressed. PTAL

File: pkg/cloudprovider/providers/gce/gce_disks_test.go (hunk 2 above)
Comment (verult) [reply pkg/cloudprovider/providers/gce/gce_disks_test.go]: Is it safe to remove the switch block entirely?

**Pull request merged**